src/manage.c

```diff
@@ -4074,20 +4074,24 @@ launch_osp_openvas_task (task_t task, target_t target, const char *scan_id,
  *
  * @param[in]   task        The task.
  * @param[in]   target      The target.
+ * @param[out]  report      The report.
  *
  * @return Parent returns with 0 if success, -1 if failure. Child process
  *         doesn't return and simply exits.
  */
 static int
-fork_osp_scan_handler (task_t task, target_t target)
+fork_osp_scan_handler (task_t task, target_t target, char **report_id)
 {
-  char *report_id, title[128], *error = NULL;
+  char *id, title[128], *error = NULL;
   int rc;
 
   assert (task);
   assert (target);
 
-  if (create_current_report (task, &report_id, TASK_STATUS_REQUESTED))
+  if (report_id == NULL)
+    report_id = &id;
```

Comment:
Overwriting the parameter seems inconsistent with how we handle NULL output pointers elsewhere.
I'd prefer to use a local variable for report_id and assigning to the output parameter where appropriate if it the pointer is not NULL.

Reply:
Used the more usual style in bd7d132.